verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -26,20 +26,17 @@ namespace verilog {
 namespace kythe {
 
 void KytheFactsExtractor::ExtractKytheFacts(const IndexingFactNode& root) {
-  CreatePackageScopes(root);
-  IndexingFactNodeTagResolver(root);
-}
-
-void KytheFactsExtractor::CreatePackageScopes(const IndexingFactNode& root) {
-  // Searches for packages as a first pass and saves their scope to be used for
-  // imports.
-  for (const IndexingFactNode& child : root.Children()) {
-    if (child.Value().GetIndexingFactType() != IndexingFactType::kPackage) {
-      continue;
-    }
-
-    VName package_vname = ExtractPackageDeclaration(child);
-    Visit(child, package_vname);
+  std::size_t number_of_extracted_facts = 0;
+  do {
```

Comment:
Add a comment above this loop. Describe the algorithm. It's not the most intuitive one.